### DISCUSSION TITLE
Closing confirmation modal only upon Promise resolution

### DISCUSSION THREAD

**Discussion by Sajarin-M** [mantinedev/mantine#3495] [Q&A] [answered]
How can I run an async function in a confirmation modal's `onConfirm` and only close the modal if the promise is resolved?
```js
onConfirm : async () => {
 try {
  await somthing();
  // close modal
 } catch (e) {
  // keep modal openened
 }
}
```
- My current solution is using `closeOnConfirm: false` and calling `closeAllModals()` inside try block. Is there a better way to achieve this?
- Calling `closeAllModals()` seems not ideal here. Is there a way to close this specific modal only?

**Comment by rtivital** [comment, marked as answer]
> My current solution is using closeOnConfirm: false and calling closeAllModals() inside try block. Is there a better way to achieve this?

It is the correct way

> Calling closeAllModals() seems not ideal here. Is there a way to close this specific modal only?

Open modal with `openConfirmModal({ modalId: 'my-id' })`, then close it `closeModal('my-id')`

**Reply by Sajarin-M** [reply]
Thanks!
